packages/common/src/index-api.ts

```diff
@@ -24,13 +24,75 @@ export type BackwardPagination = {
  */
 export type Pagination = ForwardPagination | BackwardPagination
 
+/**
+ * Boolean field value filter
+ */
+export type BooleanValueFilter = { isNull: boolean } | { equalTo: boolean }
+
+/**
+ * Enum field value filter
+ */
+export type EnumValueFilter =
+  | { isNull: boolean }
+  | { equalTo: string }
+  | { notEqualTo: string }
```

Comment:
Since we have a not filter, do we need notEqualTo or notIn?

Reply:
+1 to that.

Reply:
Counterpoint is that this isn't what is presented to users. So even though composedb/sql/etc. may say `field != value`, our internal representation doesn't need to say `notEqualTo`. This is internal or at best, part of an sdk, so needs to be able to allow us to perform the functionality, but doesn't need to be a 1 to 1 mapping. 

I prefer not having the additional operator, since it's additional code for each type we have to deal with, rather than a generalized not.

Reply:
@PaulLeCam @ukstv See above comment. I think this comes down to how much we think these changes are user facing (I don't think they are) versus the maintenance overhead of similar fields across different objects. 

Normally I'd lean towards changes that are more inline with what the user would see, but I don't think this API is very usable directly by users.

Reply:
"the maintenance overhead" - this is the most complicated part for me. If we have a general `not` instead of `notEqualTo` we would have a hard time translating that into lower level operators used in a storage backend IMO. We might expect some weird cases of deoptimisation because of generalised `not`.